releasing-archway.md

```diff
@@ -0,0 +1,111 @@
+  - These are rarely used and are special in that the X.Y.Z tag is branched to isolate
+    the emergency/critical fix from all other changes that have landed on the
+    release branch since the previous tag
+  - Cut release-X.Y.Z branch to hold the isolated patch release
+  - Tag release-X.Y.Z branch + fixes with X.Y.(Z+1)
+  - Branched [patch releases](#patch-releases) are rarely needed but used for
+    emergency/critical fixes to the latest release
+
+### Major version timeline
+
+There is currently no mandated timeline for major versions beyond version 1.Y.Z, Y,Z >= 0. We haven't so far applied a rigorous interpretation of semantic
+versioning with respect to incompatible changes of any kind.
+
+TODO: Major versioning criteria need to be put up for discussion once Archway protocol reaches 1.Y.Z release.
+
+## Patch releases
+
+Patch releases are intended for critical bug fixes to the latest minor version,
+such as addressing security vulnerabilities, fixes to problems affecting a large
+number of users and severe problems with no workaround.
+
+They should not contain miscellaneous feature additions or improvements, and
+especially no incompatibilities should be introduced between patch versions of
+the same minor version (or even major version).
+
+Dependencies, such as cosmos-sdk or tendermint, should also not be changed unless
+absolutely necessary, and also just to fix critical bugs (so, at most patch
+version changes, not new major nor minor versions).
+
+## Network Upgrades
+
+- Upgrade documentation and a graceful migration path must be provided between all **consective** minor releases.
+  e.g. going from v1.3.Z, Z >= 0 to v1.4.Z, Z >= 0 must be accompanied by relevant documentation and automated migration path.
+- Network upgrade must also follow sequential minor releases. e.g. if a network is running v1.2.Z, Z >= 0, it must first
+  upgrade to v1.3.Z, Z >= 0 before upgrading to v1.4.Z, Z >= 0
+- Network upgrades may skip patch release e.g. a network running v1.2.1 may directly upgrade to v1.2.4
+- Network upgrades on archway-1 and constantine-1 must happen through software upgrade proposals.
+- Once a software upgrade proposal passes, all network operator are expected to upgrade to the exact
+  same archway protocol release within a reasonable timeframe.
+- Each new release must be adopted and tested on constantine-1 before being adopted on archway-1.
+- titus-1 is an unstable development network, with a single validator. Network upgrades on titus-1 are
+  expected to reset network state and start again from block 1 at any given time.
+
+### Open Questions in Network upgrades:
+
+- How to build artifacts for titus-1?
+  - Should there be a build on each new commit on main branch?
+  - use X.Y.0-{alpha,beta}.W, W > 0 (Branch: master)
+    Alpha and beta releases are created from tags on main branch branch and should be used to run titus-1
+
```

Comment:
imho artifacts for Titus should be built from a `develop` branch that follows a CI/CD (Continuous Integration/Continuous Delivery) flow;
If this is strictly adhered to deployment to Titus can be automated to build and deploy from any new commit from such a branch;

Reply:
for the purposes of this proposal 'main' is that development branch. Having a second 'develop' in addition to main will fundamentally change this proposal

Reply:
But what about PR deployments?
I think it would be good to have those deployed somewhere so the makn branch would lot be this untested/verified code base.

Reply:
I think testing and verification should come in the form of integration tests. If PR deploys are still needed we can discuss a temp network setup per PR in a followup proposal